osu.Game.Rulesets.Osu/Skinning/Default/DefaultSliderBall.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osu.Game.Skinning;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Osu.Skinning.Default
+{
+    public class DefaultSliderBall : CompositeDrawable
+    {
+        private Box box;
+
+        private DrawableSlider slider;
+
+        [BackgroundDependencyLoader]
+        private void load(DrawableHitObject drawableObject, ISkinSource skin)
+        {
+            slider = (DrawableSlider)drawableObject;
+
+            RelativeSizeAxes = Axes.Both;
+
+            float radius = skin.GetConfig<OsuSkinConfiguration, float>(OsuSkinConfiguration.SliderPathRadius)?.Value ?? OsuHitObject.OBJECT_RADIUS;
+
+            InternalChild = new CircularContainer
+            {
+                Masking = true,
+                RelativeSizeAxes = Axes.Both,
+                Scale = new Vector2(radius / OsuHitObject.OBJECT_RADIUS),
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Blending = BlendingParameters.Additive,
+                BorderThickness = 10,
+                BorderColour = Color4.White,
+                Alpha = 1,
+                Child = box = new Box
+                {
+                    Blending = BlendingParameters.Additive,
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.White,
+                    AlwaysPresent = true,
+                    Alpha = 0
+                }
+            };
+
+            slider.Tracking.BindValueChanged(trackingChanged, true);
+            slider.ApplyCustomUpdateState += updateStateTransforms;
+        }
+
+        private void trackingChanged(ValueChangedEvent<bool> tracking) =>
+            box.FadeTo(tracking.NewValue ? 0.3f : 0.05f, 200, Easing.OutQuint);
+
+        private void updateStateTransforms(DrawableHitObject obj, ArmedState state)
+        {
+            if (!(obj is DrawableSlider))
+                return;
+
+            const float fade_out_time = 112.5f;
+
+            using (BeginAbsoluteSequence(slider.HitObject.StartTime))
+                this.FadeIn().ScaleTo(1f);
+
+            using (BeginAbsoluteSequence(slider.HitObject.EndTime))
+            {
+                this.FadeOut(fade_out_time, Easing.Out);
+                switch (state)
+                {
+                    case ArmedState.Hit:
+                        this.ScaleTo(1.1f, fade_out_time, Easing.Out);
```

Comment:
Is this supposed to be the scale transform that was in `DrawableSlider`? If so, why has the final value changed (from `HitObject.Scale * 1.4f`)?

Reply:
The `fade_out_time` is 1/4 of what it used to be. so the scale to animation is also 1/4 of what it was (10% instead of 40%). Before, `fade_out_time` was misleading because the fade out animation was applied with `fade_out_time/4`, while the full `fade_out_time` was only used for the scale out animation. So really, the old `fade_out_time` was more like a scale out time.

Reply:
`Easing.Out` [is not a linear easing function](https://github.com/ppy/osu-framework/blob/269c6c63625e56ec269dcf7024fb0e95baf68d36/osu.Framework/Graphics/Transforms/DefaultEasingFunction.cs#L50-L52). So you can't quarter the final value and the duration and expect to get the same result.

Please just restore these values to what they were originally.

Reply:
Derp, I'm dumb. Restored.